src/texdocdialog.cpp

```diff
@@ -193,7 +195,19 @@ void TexdocDialog::checkDockAvailable()
     if (lastDocRequest.isEmpty()){
         updateDocAvailableInfo("", false);
     } else {
-        help->texdocAvailableRequest(lastDocRequest);
+        LatexRepository *repo = LatexRepository::instance();
+        if (repo->LatexRepository::packageExists(lastDocRequest)) {
+            TeXdocStatus status = repo->LatexRepository::docStatus(lastDocRequest);
+            if (status == Undefined)
+                help->texdocAvailableRequest(lastDocRequest);
+            else {
+                QString message = repo->LatexRepository::docMessage(lastDocRequest);

```

Comment:
is there actually a case where this message gets set to something ? Looks like circular ?

Reply:
When a row is selected the first time (state is Undefined), state and message are retrieved and stored in the latexPackageInfo (s. line 201f in `TexdocDialog::regenerateTable` and changes in `TexdocDialog::updateDocAvailableInfo`).
If the message would not be set, then you would not see any message in case you select again the row with a package that has no TeXdoc available.